src/Uno.Extensions.Core.Generators/PropertySelector/PropertySelectorCandidate.cs

```diff
@@ -1,43 +1,201 @@
-		var position = syntax is null ? default : syntax.SyntaxTree.GetLineSpan(syntax.Span);
-		var method = syntax is null
-			? null
-			: context.SemanticModel.GetSymbolInfo(syntax, ct).Symbol as IMethodSymbol;
-		var hasPropertySelector = method?.Parameters.Any(p => p.Type is INamedTypeSymbol
-		{
-			IsGenericType: true,
-			MetadataName: "PropertySelector`2",
-			ContainingNamespace: { Name: "Edition", ContainingNamespace: { Name: "Extensions", ContainingNamespace.Name: "Uno" } }
-		});
-
-		Context = context;
-		Syntax = syntax;
-		Location = position;
-		Method = method;
-		IsValid = hasPropertySelector ?? false;
-	}
+		Syntax = (InvocationExpressionSyntax)context.Node;
+		Location = Syntax.SyntaxTree.GetLineSpan(Syntax.Span);
+		
+		var method = context.SemanticModel.GetSymbolInfo(Syntax, ct).Symbol as IMethodSymbol;
+		
+		if (method is null ||
+			!method.Parameters.Any(p => p.Type is INamedTypeSymbol
+			{
+				IsGenericType: true,
+				MetadataName: "PropertySelector`2",
+				ContainingNamespace: { Name: "Edition", ContainingNamespace: { Name: "Extensions", ContainingNamespace.Name: "Uno" } }
+			}) ||
+			method.Parameters.FirstOrDefault(p => p.HasAttribute<CallerFilePathAttribute>()) is not { } callerPathParameter ||
+			method.Parameters.FirstOrDefault(p => p.HasAttribute<CallerLineNumberAttribute>()) is not { } callerLineParameter
+			)
+		{
+			Accessors = null;
+			MethodGlobalNamespace = null;
+			MethodName = null;
+			return;
+		}
+
+		var arguments = Syntax.ArgumentList.Arguments;
+
+		var callerPathArg = arguments.ElementAtOrDefault(callerPathParameter.Ordinal)?.Expression;
+		var callerPath = callerPathArg switch
+		{
+			LiteralExpressionSyntax pathArg => pathArg.Token.ValueText,
+			// It might be something else, but in this case the analyzer will complain.
+			_ => Location.Path,
+		};
 
-	[MemberNotNullWhen(true, nameof(Syntax), nameof(Method))]
-	public bool IsValid { get; }
+		var callerLineArg = arguments.ElementAtOrDefault(callerLineParameter.Ordinal)?.Expression;
+		var callerLine = callerLineArg switch
+		{
+			LiteralExpressionSyntax lineArg => lineArg.Token.ValueText,
+			// It might be something else, but in this case the analyzer will complain.
+			_ => (Location.StartLinePosition.Line + 1).ToString(CultureInfo.InvariantCulture),
+		};
+
+		MethodName = method.Name;
+		MethodGlobalNamespace = method.ContainingModule?.GlobalNamespace?.ToString() ?? "";
 
-	public GeneratorSyntaxContext Context { get; }
+		Accessors = method.Parameters
+			.Where(param => param.Type is INamedTypeSymbol
+			{
+				IsGenericType: true,
+				MetadataName: "PropertySelector`2",
+				ContainingNamespace: { Name: "Edition", ContainingNamespace: { Name: "Extensions", ContainingNamespace.Name: "Uno" } }
+			})
+			.Select(param =>
+				(key: $"{param.Name}{callerPath}{callerLine}", accessor: GenerateAccessor((INamedTypeSymbol)param.Type, arguments.ElementAtOrDefault(param.Ordinal))))
```

Comment:
Could / should we defer the generation of the code until really needed?

Reply:
@dr1rrb I guess it will always be needed. When we reach here, that means it is a valid candidate, which means we will be calling `Generate` which will need `Accessors`